### ISSUE TITLE
Add ban/unban and approve/reject methods to User

### ISSUE THREAD

**Issue by sathieu** [python-gitlab/python-gitlab#1604]
## Description of the problem, including code/CLI snippet

API has endpoints to manage the state of a user:

- activate/deactivate: :heavy_check_mark: already done
- block/unblock: :heavy_check_mark: already done
- ban/unban: :construction: TODO
- approve/reject: :construction: TODO

**Comment by nejch** [comment]
@antoineauger as discussed if you want a break from k8s and work on this :grin: 

There's a PR that implements approve/reject endpoints for this issue in https://github.com/python-gitlab/python-gitlab/pull/2061 - you could do the same for ban/unban:
https://docs.gitlab.com/ee/api/users.html#ban-user
https://docs.gitlab.com/ee/api/users.html#unban-user

See my comments in the other PR for a bit of guidance, although ideally we'd also have 1 response per mock and 1 assert per test so we can remove `assert_all_requests_are_fired=False`.

In the case of ban/unban I think we could also add a quick functional test like for block/unblock. https://github.com/python-gitlab/python-gitlab/blob/main/tests/functional/api/test_users.py#L30-L37.

With both PRs in we could then close this issue. Of course if there's any hiccups along the way I'm happy to look at docs PRs as well if you feel like it :grinning:
